src/models.zig

```diff
@@ -149,6 +161,296 @@ pub const Model = struct {
+			if (lineUntrimmed.len < 3)
+				continue;
+			
+			var line = lineUntrimmed;
+			if (line[line.len - 1] == '\r') {
+				line = line[0..line.len - 1];
+			}
+			
+			if (std.mem.eql(u8, line[0..1], "#"))
+				continue;
+			
+			if (std.mem.eql(u8, line[0..2], "v ")) {
+				var coordsIter = std.mem.split(u8, line[2..], " ");
+				var coords: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					coords[i] = try std.fmt.parseFloat(f32, coord);
+				}
+				const coordsCorrect: Vec3f = .{coords[0] + 0.5, -coords[2] + 0.5, coords[1]};
+				try vertices.append(coordsCorrect);
+			} else if (std.mem.eql(u8, line[0..3], "vn ")) {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var norm: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					norm[i] = try std.fmt.parseFloat(f32, coord);
+				}
+				const normCorrect: Vec3f = .{norm[0], -norm[2], norm[1]};
+				try normals.append(normCorrect);
+			} else if (std.mem.eql(u8, line[0..3], "vt ")) {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var uv: Vec2f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					uv[i] = try std.fmt.parseFloat(f32, coord);
+				}
+				uv[0] *= 4;
+				uv[1] *= 4;
+				try uvs.append(.{uv[0], uv[1]});
+			} else if (std.mem.eql(u8, line[0..2], "f ")) {
+				if (std.mem.count(u8, line[2..], " ") + 1 == 3) {
+					var coordsIter = std.mem.split(u8, line[2..], " ");
+					var faceData: [3][3]usize = undefined;
+					var i: usize = 0;
+					while (coordsIter.next()) |vertex| : (i += 1) {
+						var d = std.mem.split(u8, vertex, "/");
+						var j: usize = 0;
+						while (d.next()) |value| : (j += 1) {
+							faceData[j][i] = try std.fmt.parseUnsigned(usize, value, 10) - 1;
+						}
+					}
+					try tris.append(.{.vertex=faceData[0], .uvs=faceData[1], .normals=faceData[2]});
+				} else {
+					var coordsIter = std.mem.split(u8, line[2..], " ");
+					var faceData: [3][4]usize = undefined;
+					var i: usize = 0;
+					while (coordsIter.next()) |vertex| : (i += 1) {
+						var d = std.mem.split(u8, vertex, "/");
+						var j: usize = 0;
+						while (d.next()) |value| : (j += 1) {
+							faceData[j][i] = try std.fmt.parseUnsigned(usize, value, 10) - 1;
+						}
+					}
+					try quadFaces.append(.{.vertex=faceData[0], .uvs=faceData[1], .normals=faceData[2]});
+				}
+			}
+		}
+
+		var quadInfos = std.ArrayList(QuadInfo).init(main.stackAllocator.allocator);
+		defer quadInfos.deinit();
+
+		for (tris.items) |face| {
+			const normal: Vec3f = normals.items[face.normals[0]];
+			
+			var uvA: Vec2f = uvs.items[face.uvs[0]];
+			var uvB: Vec2f = uvs.items[face.uvs[1]];
+			var uvC: Vec2f = uvs.items[face.uvs[2]];
+
+			
+			var minUv = @floor(@min(@min(uvA, uvB), uvC));
+			const textureSlot = @as(u32, @intFromFloat(@floor(minUv[1]))) * 4 + @as(u32, @intFromFloat(@floor(minUv[0])));
```

Comment:
What happens if the minUv is larger than 4.0? What if they are smaller than 0.0? Cases like these need to be handled.

Reply:
this could only happen if the uv is outside of the 0-1 range

Reply:
Which is something that absolutely can happen.

Reply:
now it wont work if the uvs are outside of 0-1 and it will print an error